app/controllers/application_controller.rb

```diff
@@ -20,4 +20,8 @@ def configure_permitted_parameters
   helper_method memoize def goal_percentage(account, objective)
     ActionController::Base.helpers.number_to_percentage(account.balance / objective.amount * 100, precision: 0)
   end
+
+  helper_method def canonical_path
+    (url_for(only_path: true) if [:home, :policy].include?(params[:controller])) || ''
+  end
```

Comment:
```suggestion
  helper_method def canonical_url
    "https://budgetingkid.com#{request.path}"
  end
```

Reply:
We will handle all other pages with query parameters, if necessary, separately. It's barely possible though. I can't imagine a scenario when we need to index pages with query params.

Reply:
In light of this PR/discussion:
- using "https://example.com#{url_for(only_path: true)}" is more reliable;
- full current path of a private page as the canonical url in html isn't faultless job of mine
- both it's not essential
above all, let('s) move on or run

Reply:
Why it is more reliable?

Reply:
`url_for` is more reliable then `request.path` because it's created on a controller attributes - this is a controlled process, what will be in the request we do not know.

Reply:
As for me they are the same. The only difference is that url for builds path from scratch and that takes some time. So it's less performant. On the other hand, if no route defined for the requested path it will never get to this line of code. That would be great to hear more detailed explanation why building path dynamically is more reliable. Maybe some tests would show that?